### PR TITLE
Refactor: connection load test with different IP sources

### PR DESCRIPTION
**Pull request by asmie**
Connection load test now can use different source IPs. Ported assertions from zcash to check test results.

Seems that rippled still does not behave like it should accepting more connections than max_peers and terminating some other connections without a clear reason.

Currently it's recommended to generate addresses to cover all iterations of all tests, so it would be good to generate whole 24bit subnet. When pea2pea will allow to set REUSE_ADDR option "the hack" can be removed.

### CODE REVIEW

**Comment by lwojtow** [comment]
Approved, but I guess this will be changed anyway when Lukasz makes his changes to pea2pea.

**Review by lwojtow — APPROVED** [review]

File: src/tests/performance/connections.rs [hunk 1]
```diff
@@ -173,16 +246,48 @@ async fn p002_connections_load() {
...
+
+        // And no connection timeouts or errors
+        assert_eq!(stats.timed_out, 0, "Stats: {:?}", stats);
+        assert_eq!(stats.conn_error, 0, "Stats: {:?}", stats);
+    }
 }
 
-async fn simulate_peer(node_addr: SocketAddr, handshake_complete: Sender<()>) {
-    let config = TestConfig::default();
+async fn simulate_peer(node_addr: SocketAddr, handshake_complete: Sender<()>, thread_num: usize) {
```
Comment (Rqnsom) [inline src/tests/performance/connections.rs:277]: we could also rename `thread_num` to something else here

**Review by Rqnsom — APPROVED** [review]
LGTM

File: src/tests/performance/connections.rs (hunk 1 above)
Comment (asmie) [reply src/tests/performance/connections.rs]: Changed.

**Pull request merged**